Sources/PackageDescription4/JSON.swift

```diff
@@ -46,4 +46,399 @@ extension JSON {
             return output
         }
     }
+
+    var isArray: Bool {
+        if case .array = self {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    var array: [JSON]? {
+        get {
+            if case .array(let array) = self {
+                return array
+            } else {
+                return nil
+            }
+        }
+        set {
+            guard let array = newValue else {
+                fatalError("expected value")
+            }
+
+            self = .array(array)
+        }
+    }
+
+    var isDictionary: Bool {
+        if case .dictionary = self {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    var dictionary: [String: JSON]? {
+        get {
+            if case .dictionary(let dictionary) = self {
+                return dictionary
+            } else {
+                return nil
+            }
+        }
+        set {
+            guard let dictionary = newValue else {
+                fatalError("expected value")
+            }
+
+            self = .dictionary(dictionary)
+        }
+    }
+}
+
+final class JSONEncoder {
+    init() {
+    }
+
+    func encode<T: Encodable>(_ value: T) throws -> String {
+        let encoder = _JSONEncoder()
```

Comment:
Why create this every time encode is called?

Reply:
To make sure we start with a clean state every time we encode a new value. It mirrors what the official `JSONEncoder` does: https://github.com/apple/swift/blob/master/stdlib/public/SDK/Foundation/JSONEncoder.swift#L212